LEGO1/omni/src/action/mxdsaction.cpp

```diff
@@ -258,24 +258,27 @@ void MxDSAction::AppendExtra(MxU16 p_extraLength, const char* p_extraData)
 // FUNCTION: BETA10 0x1012ba6a
 void MxDSAction::Deserialize(MxU8*& p_source, MxS16 p_flags)
 {
+	double tempDouble;
+
 	MxDSObject::Deserialize(p_source, p_flags);
 
 	// clang-format off
-	m_flags           = *( MxU32*) p_source;  p_source += sizeof(m_flags);
-	m_startTime       = *(MxLong*) p_source;  p_source += sizeof(m_startTime);
-	m_duration        = *(MxLong*) p_source;  p_source += sizeof(m_duration);
-	m_loopCount       = *( MxS32*) p_source;  p_source += sizeof(m_loopCount);
-	m_location[0]     = *(double*) p_source;  p_source += sizeof(double);
-	m_location[1]     = *(double*) p_source;  p_source += sizeof(double);
-	m_location[2]     = *(double*) p_source;  p_source += sizeof(double);
-	m_direction[0]    = *(double*) p_source;  p_source += sizeof(double);
-	m_direction[1]    = *(double*) p_source;  p_source += sizeof(double);
-	m_direction[2]    = *(double*) p_source;  p_source += sizeof(double);
-	m_up[0]           = *(double*) p_source;  p_source += sizeof(double);
-	m_up[1]           = *(double*) p_source;  p_source += sizeof(double);
-	m_up[2]           = *(double*) p_source;  p_source += sizeof(double);
-
-	MxU16 extraLength = *( MxU16*) p_source;  p_source += sizeof(extraLength);
+	memcpy(&m_flags,     p_source, sizeof(MxU32));  p_source += sizeof(MxU32);
```

Comment:
Maybe instead of using memcpy everywhere we could add a small template utility:

```
template <typename T>
T UnalignedRead(MxU8* p_source) {
    T value;
    memcpy(&value, p_source, sizeof(T));
    return value;
}
```
(untested)
I think it would make the code a bit more readable, and no need for extra temporary vars in some cases

Reply:
Where would you like me to put it? I did consider it but dropped it once I had a version of it in 10 files.

Reply:
hmm, I'd say either:

https://github.com/isledecomp/isle-portable/blob/master/LEGO1/omni/include/mxutilities.h

it already has similar functions.

or this one:

https://github.com/isledecomp/isle-portable/blob/master/util/compat.h

since it contains "compatibility" related things.

@madebr wdyt

Reply:
I'd put it in `mxutilities.h`. I think this is supported by early c++ compilers?